### PR TITLE
CI: Add a PR check to do a stand-alone kernel compilation 

### PR DESCRIPTION
**Pull request by LukeMondy**
On a PR, compile all verified configs of seL4.

I put it in its own yml file, so we can better sort through the results, using the GitHub GUI.

### CODE REVIEW

File: .github/workflows/compilation-checks.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Compilation actions to run on pull requests
+
+name: Compilation checks
+
+on: [pull_request]
```
Comment (lsf37) [inline .github/workflows/compilation-checks.yml:9]:
```suggestion
on:
  push:
    branches:
      - master
  pull_request:
```

I'd run these on push to master as well. Theoretically it should already be implied that this always works on master if we check on every PR, but it's nice to be able to look at just the branch and directly see the CI status.

File: .github/workflows/compilation-checks.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Compilation actions to run on pull requests
+
+name: Compilation checks
```
Comment (lsf37) [inline .github/workflows/compilation-checks.yml:7]: Can we find a shorter name? Maybe just "Compile". The shorter the name, the easier it is to see what's happening in the "Checks" section of a PR, esp with matrix builds that tend to generate long combination names.

Comment (LukeMondy) [reply .github/workflows/compilation-checks.yml:7]: Yep, `Compile` sounds good.

Looking to the future though, the number of checks for `sel4test` would be >>50. So perhaps we should isolate them, meaning this one would be `Kernel compile`, and the future seL4test one would be `seL4test compile`.

Comment (lsf37) [reply .github/workflows/compilation-checks.yml:7]: We could also call the workflow `Kernel`, and the job `compile`. Then it should come out as something like `Kernel / compile (ARM, llvm, py3)`.

Comment (lsf37) [reply .github/workflows/compilation-checks.yml:7]: But happy for you to decide either way.

**Comment by lsf37** [comment]
Hm. The DCO check complains about commits made via the `suggestion` interface. Happy to for you to just set it to `pass` for this one (trivial change).

**Comment by LukeMondy** [comment]
I'm not actually logged on to my work VM, so I can't rebase it now (or make the name changes).
Happy for you to do it now, or I'll do it tomorrow (so it passes the DCO, and with the suggested name changes)

**Comment by lsf37** [comment]
I'll give it a go

File: .github/workflows/compilation-checks.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (lsf37) [inline .github/workflows/compilation-checks.yml:7]:
```suggestion
name: Kernel
```

File: .github/workflows/compilation-checks.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  standalone_kernel:
+    name: Compile kernel
```
Comment (lsf37) [inline .github/workflows/compilation-checks.yml:17]:
```suggestion
    name: compile
```

**Comment by lsf37** [comment]
Squashed the commits -- I hope this worked ;)

**Comment by lsf37** [comment]
So this didn't work quite as nicely as I expected. The squash showed me as the commit author (I thought), hence I added Luke as co-authored-by, but it turns out Luke is there now twice. And the commit message is confusing, because I missed the subject line of the interface. I should do a proper manual rebase/squash next time -- sorry about the borked commit.

**Pull request merged**